### ISSUE TITLE
Cannot enter decimal places

### ISSUE THREAD

**Issue by spaceChRiS** [inguin/moneybalance#21]
I have a phone with German localization, Android 5.0 and the standard UI enhancements by Samsung. To enter an expense, I am presented with a numerical keyboard. This keyboard only allows to enter a “.” as decimal separator, and it cannot be replaced by another keyboard. When entering decimal places, moneybalance indicates that I did not enter a valid number and does not let me save it. So I have no possibility to enter decimal places. When I paste a number separated by “,” (comma) into the field (pasting is possible), it does not complain. To circumvent such problems, I suggest that “.” and “,” are equally treated as decimal separator and are automatically translated to the format required by the database or the localization settings.

**Comment by inguin** [comment]
Interesting, must be a Samsung "feature". I'm explicitly requesting a keyboard with both period and comma keys, and it works as expected on my Nexus 4 (plain Google Android 5.1.1, German locale).

I'll see what I can do.

**Comment by jschaeff** [comment]
There is also a similar problem with the CSV output. The google sheet does not understand the data as decimal numbers but as strings. I had to replace the dots with comas manually to make it work.

**Comment by inguin** [comment]
Hmm, works for me again. Google Sheets recognizes all amount cells as numbers, even though the CSV uses a point as decimal separator and my system is configured for German language.

What's your version of the Sheets app? I have 1.6.192.07.34. Have you tried it with a different calculation? Maybe they somehow guess which decimal separator to use, and there's something in your data that throws them off.

**Comment by inguin** [comment]
Unfortunately I could not come up with a good solution for this yet. The problem is: When editing an existing expense, the amount is formatted according to the locale settings. For instance, with a German locale the contents of the input field could be 1.234,00. A user might change that value and remove the decimal part, e.g. 1.235. Now, if I treat "." as decimal separator this will be misinterpreted.

**Comment by spaceChRiS** [comment]
That's a pity. And what about having a configuration option to enter cents instead of euros (or whatever the currencies are). That way I could type 1500 and it would be saved as 15,00. The same as supermarket cash tills work. That's what I do due to this issue anyway, but having 100 € instead of 1 € in the list confuses and leads to wrong entries.

**Comment by inguin** [comment]
Yup, right now introducing a settings screen seems like the safest way to deal with this issue without breaking anybody else's experience. Of course the option wouldn't be about entering Cents instead of Euros, but rather overriding the locale-defined decimal separator.
